utils/checkRoles.js

```diff
@@ -0,0 +1,58 @@
+const { readFileSync } = require('fs');
+const { Client, GatewayIntentBits, EmbedBuilder } = require('discord.js');
+
+const getChannel = (discordClient, channelName) => {
+  return (discordClient.channels.cache).find((channel) => channel.name === channelName);
+};
+
+const rolesChannel = "roles-on-chain"
+
+const client = new Client({ intents: [GatewayIntentBits.Guilds, GatewayIntentBits.GuildMessages] });
+
+client.on('ready', async () => {
+  console.log(`Logged in as ${client.user.tag}!`);
+
+  const channel = getChannel(client, rolesChannel);
+  if (!channel) {
+    console.log('discord channel not found');
+    return;
+  }
+
+  const content = readFileSync('./scripts/roles.json');
+  const roles = JSON.parse(content);
+  const chains = Object.keys(roles);
+  let embeds = [];
```

Comment:
embeds will be populated with all chains such that for the last chain on l.45 we are going to go through all embeds even the one not attached to the current chain